jerry-core/ecma/builtin-objects/ecma-builtin-arraybuffer-prototype.c

```diff
@@ -0,0 +1,149 @@
+#include "ecma-builtins.h"
+#include "ecma-conversion.h"
+#include "ecma-exceptions.h"
+#include "ecma-function-object.h"
+#include "ecma-gc.h"
+#include "ecma-globals.h"
+#include "ecma-helpers.h"
+#include "ecma-objects.h"
+#include "ecma-arraybuffer-object.h"
+#include "ecma-try-catch-macro.h"
+#include "jrt.h"
+#include "jrt-libc-includes.h"
+
+#ifndef CONFIG_DISABLE_ARRAYBUFFER_BUILTIN
+
+#define ECMA_BUILTINS_INTERNAL
+#include "ecma-builtins-internal.h"
+
+#define BUILTIN_INC_HEADER_NAME "ecma-builtin-arraybuffer-prototype.inc.h"
+#define BUILTIN_UNDERSCORED_ID arraybuffer_prototype
+#include "ecma-builtin-internal-routines-template.inc.h"
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmabuiltins
+ * @{
+ *
+ * \addtogroup arraybuffer prototype ECMA ArrayBuffer.prototype object built-in
+ * @{
+ */
+
+/**
+ * The ArrayBuffer.prototype.bytelength accessor
+ *
+ * See also:
+ *          ES2015, 24.1.4.1
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_arraybuffer_prototype_bytelength_getter (ecma_value_t this_arg)
+{
+  if (ecma_is_value_object (this_arg))
+  {
+    ecma_object_t *object_p = ecma_get_object_from_value (this_arg);
+    if (ecma_object_class_is (object_p, LIT_MAGIC_STRING_ARRAY_BUFFER_UL))
+    {
+      ecma_length_t len = ecma_arraybuffer_get_length (object_p);
+
+      return ecma_make_uint32_value (len);
+    }
+
+  }
+  return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a ArrayBuffer object."));
+} /* ecma_builtin_arraybuffer_prototype_bytelength_getter */
+
+/**
+ * The ArrayBuffer.prototype object's 'slice' routine
+ *
+ * See also:
+ *          ES2015, 24.1.4.3
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_arraybuffer_prototype_object_slice (ecma_value_t this_arg, /**< this argument */
+                                                 ecma_value_t arg1, /**< routine's first argument */
+                                                 ecma_value_t arg2) /**< routine's second argument */
+{
+  if (!ecma_is_value_object (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not object."));
+  }
+
+  ecma_object_t *object_p = ecma_get_object_from_value (this_arg);
+
+  if (!ecma_object_class_is (object_p, LIT_MAGIC_STRING_ARRAY_BUFFER_UL))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not an ArrayBuffer object."));
+  }
+
+  ecma_length_t len = ecma_arraybuffer_get_length (object_p);
+
+  ecma_length_t start = 0, end = len;
+
+  ecma_value_t ret_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+
+  ECMA_OP_TO_NUMBER_TRY_CATCH (start_num,
+                               arg1,
+                               ret_value);
+
+  start = ecma_builtin_helper_array_index_normalize (start_num, len);
+
+  if (ecma_is_value_undefined (arg2))
+  {
+    end = len;
+  }
```

Comment:
end is initialized above. Do we need this?

ecma_length_t start = 0, end = len;